src/ActiveLogin.Authentication.BankId.AspNetCore/Areas/BankIdAuthentication/Controllers/BankIdApiController.cs

```diff
@@ -273,7 +274,7 @@ private async Task<ActionResult> CollectComplete(BankIdLoginApiStatusRequest req
                 throw new ArgumentNullException(nameof(request.ReturnUrl));
             }
 
-            _logger.BankIdCollectCompleted(collectResponse.OrderRef, collectResponse.CompletionData);
+            _bankIdEventTrigger.TriggerAsync(new BankIdCollectCompletedEvent(collectResponse.OrderRef, collectResponse.CompletionData));
             foreach (var bankIdResultStore in _bankIdResultStores)
```

Comment:
The `IBankIdResultStore` will eventually be obsolete, because people can listen to the `BankIdCollectCompletedEvent` event instead. For now I would vote keeping it to be backwards compatible. But, I would move the `foreach (var bankIdResultStore in _bankIdResultStores)`-loop into a class that implements IBankIdEventListener instead. It would make it cleaner for now.

Reply:
Created a new ResultStoreBankIdEventListener and initialized it from AddDefaultServices().